File: Mock/Data/Internal/DataType.cs
namespace Mock.Data
{
    using System;
    using System.Reflection;

    using Mock.Data.Attributes;
    using Mock.Data.Exception;
    internal class DataType : TaistType
    {
        public override string FriendlyName
        {
            get
            {
                object[] typeAttributeArray = reflectDataType.GetCustomAttributes(typeof(DataClassAttribute), false);
                if (typeAttributeArray.Length > 0)
                {
                    DataClassAttribute classAttribute = typeAttributeArray[0] as DataClassAttribute;
                    if (string.IsNullOrEmpty(classAttribute.Name))
                    {
                        return Name;
                    }
                    return classAttribute.Name;
                }
                else
                {
                    return Name;
                }
            }
        }
    }
}

File: java/Taist.Framwork/src/Mock/Data/Internal/DataType.java
package Mock.Data.Internal;

import Mock.Data.Annotations.DataClass;
import Mock.Data.Helpers.StringHelper;

class DataType extends AbstractTaistType {
	public DataType(Class<?> realType) {
		super(realType);
	}
	
	@Override
	public String getFriendlyName() {
		DataClass[] dataAnnotations = realType.getAnnotationsByType(DataClass.class);
		if(dataAnnotations != null && dataAnnotations.length > 0) {
			String friendlyName = dataAnnotations[0].Name();
			return StringHelper.isNullOrEmpty(friendlyName) ? 
					super.getFriendlyName() : friendlyName;
		}
		else {
			return super.getFriendlyName();
		}
	}
}
